src/components/auth/Login.jsx

```diff
@@ -1,14 +1,93 @@
+import { AuthProvider, useAuth } from "../../AuthProvider";
+import { Link, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
 import FormInput from "../common/FormInput";
+import { login } from "../../util/fetchData"
+
 const Login = () => {
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [formErrors, setFormErrors] = useState({});
+  const { isLoggedIn, loginAction } = useAuth();
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+    setFormErrors({ ...formErrors, [e.target.name]: '', form: '' });
+  }
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!formData.email || !formData.password) {
+      setFormErrors({
+        ...formErrors,
+        email: !formData.email ? "Email is required" : null,
+        password: !formData.password ? "Password is required" : null,
+      });
+      return;
+    }
+
+    try {
+      const result = await login(formData);
+      if (result.status === 200) {
+        let userData = result.data.user;
+        loginAction(true, userData);
+        navigate("/dashboard");
+      }
+    } catch (error) {
+      setFormErrors({
+        ...formErrors,
+        form: error.message,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      navigate("/dashboard");
+    }
+  }, [isLoggedIn, navigate]);
```

Comment:
I can see that the intention here is for a user to be logged out when ever their log in status is changed, this is good. What is the reason to include navigate as a dependency?